feature/preview/src/main/java/com/google/jetpackcamera/feature/preview/ui/PreviewScreenComponents.kt

```diff
@@ -365,3 +387,164 @@ fun CaptureButton(
+    val circleColor = if (enabled) circleSelectionColor else disableColor.copy(alpha = 0.12f)
+    var toggleState by remember { mutableStateOf(initialState) }
+    val animatedTogglePosition by animateFloatAsState(
+        when (toggleState) {
+            ToggleState.Left -> 0f
+            ToggleState.Right -> 1f
+        },
+        label = "togglePosition"
+    )
+    val scope = rememberCoroutineScope()
+
+    Surface(
+        modifier = modifier
+            .clip(shape = RoundedCornerShape(50))
+            .then(
+                if (enabled) {
+                    Modifier.clickable {
+                        scope.launch {
+                            toggleState = when (toggleState) {
+                                ToggleState.Left -> ToggleState.Right
+                                ToggleState.Right -> ToggleState.Left
+                            }
+                            onToggleStateChanged(toggleState)
+                        }
+                    }
+                } else {
+                    Modifier
+                }
+            ),
+        color = backgroundColor
+    ) {
+        Box {
+            Row(
+                modifier = Modifier.matchParentSize(),
+                verticalAlignment = Alignment.CenterVertically
+            ) {
+                Box(
+                    Modifier
+                        .layout { measurable, constraints ->
+                            val placeable = measurable.measure(constraints)
+                            layout(placeable.width, placeable.height) {
+                                val xPos = animatedTogglePosition *
+                                    (constraints.maxWidth - placeable.width)
+                                placeable.placeRelative(xPos.toInt(), 0)
+                            }
+                        }
+                        .fillMaxHeight()
+                        .aspectRatio(1f)
+                        .clip(RoundedCornerShape(50))
+                        .background(circleColor)
+                )
+            }
+            Row(
+                modifier = Modifier.matchParentSize().then(
+                    if (enabled) Modifier else Modifier.alpha(0.38f)
+                ),
+                verticalAlignment = Alignment.CenterVertically,
+                horizontalArrangement = Arrangement.SpaceBetween
+            ) {
+                Icon(
+                    painter = leftIcon,
+                    contentDescription = "leftIcon",
+                    modifier = Modifier.padding(iconPadding),
+                    tint = if (!enabled) {
+                        disableColor
+                    } else if (toggleState == ToggleState.Left) {
+                        iconSelectionColor
+                    } else {
+                        iconUnSelectionColor
+                    }
+                )
+                Icon(
+                    painter = rightIcon,
```

Comment:
Add this content discription as an argument to the composable with the default of `"rightIcon"`. We should be able to change it to be an accurate description of the icon.

Reply:
Thank you, done.